lib/matplotlib/widgets.py

```diff
@@ -2099,6 +2105,9 @@ def __init__(self, ax, onselect, direction, minspan=0, useblit=False,
         self.grab_range = grab_range
         self._interactive = interactive
         self.drag_from_anywhere = drag_from_anywhere
+        self.ignore_event_outside = ignore_event_outside
+
+        self._completed = False
```

Comment:
I guess this could be renamed to something like `_last_selection_succeeded` or at least could get a comment explaining what it is?

Reply:
Rename to `_completed` to `_selection_completed` and added comments in the base class.